src/notifications.js

```diff
@@ -71,12 +71,49 @@ export default class Notification {
     }
   }
 
+  mapCombineTopics (topics) {
+    const prefixes = {};
+    topics.map(t => {
+      const split = t.split('.');
+      const postfix = split.splice(split.length - 1);
+      const prefix = split.join('.');
+      return { prefix, postfix };
+    }).forEach(t => {
+      if (prefixes[t.prefix]) {
+        prefixes[t.prefix].push(t.postfix);
+      } else {
+        prefixes[t.prefix] = [ t.postfix ];
+      }
+    });
+
+    let combinedTopics = [];
+
+    // Max length of 200 in topic names
+    // so recursively break them up if the combined length exceeds 200
+    const combineTopics = function (prefix, postFixes) {
+      const delimiter = postFixes.length === 1 ? '.' : '?';
+      const id = `${prefix}${delimiter}${postFixes.join('&')}`;
+      if (id.length < 200) {
+        combinedTopics.push({ id });
+      } else {
```

Comment:
Would it ever be possible to have id.length >= 200 and only one postfix? if so, this would loop forever right?

Reply:
Fixed

Reply:
Fixed